### PR TITLE
Refactoring suggestion

### PR DESCRIPTION
**Pull request by laurentqro**

### CODE REVIEW

File: app/models/legacy/server.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+class Legacy::Server
+  def update(payload)
+    RestClient.put(url, payload, headers)
```
Comment (govuklaurence) [inline app/models/legacy/server.rb]: Log if the PUT fails

File: app/models/legacy/dataset.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
+class Legacy::Dataset < SimpleDelegator
+  def json_metadata
+    {
+      'id': uuid,
```
Comment (govuklaurence) [inline app/models/legacy/dataset.rb]: this line will have a symbol key

File: app/models/legacy/server.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  end
+
+  private
+
+  def url
+    "https://#{host}/#{path}"
```
Comment (govuklaurence) [inline app/models/legacy/server.rb]: Use `URI::join`

**Comment by hannako** [comment]
Closing this PR so I can delete branch. I will implement this refactor on my branch and open a fresh PR

**Pull request closed without merging**